### PR TITLE
cow: optimization of pages in backing objects

### PR DESCRIPTION
**Pull request by bkjg**
This PR depends on #832

### CODE REVIEW

File: sys/kern/vm_object.c [hunk 1]
```diff
@@ -36,37 +36,40 @@ void vm_object_add_page(vm_object_t *obj, off_t offset, vm_page_t *pg) {
...
+      obj->npages++;
+      return;
+    }
+    /* there must be no page at the offset! */
+    assert(it->offset != pg->offset);
   }
+
+  /* offset of page is greater than the offset of any other page */
+  TAILQ_INSERT_TAIL(&obj->list, pg, obj.list);
+  obj->npages++;
```
Comment (wiklam) [inline sys/kern/vm_object.c:56]: Something changed here or WITH_MTX_LOCK was replaced by SCOPED_MTX_LOCK? Please don't do that!!! It waste just time when someone wants to read it.

Comment (bkjg) [reply sys/kern/vm_object.c]: Fixed.

File: sys/aarch64/pmap.c [hunk 2]
```diff
@@ -511,24 +511,34 @@ pmap_t *pmap_new(void) {
...
-  while (!TAILQ_EMPTY(&pmap->pte_pages)) {
-    vm_page_t *pg = TAILQ_FIRST(&pmap->pte_pages);
-    TAILQ_REMOVE(&pmap->pte_pages, pg, pageq);
-    vm_page_free(pg);
-  }
+    while (!TAILQ_EMPTY(&pmap->pte_pages)) {
+      vm_page_t *pg = TAILQ_FIRST(&pmap->pte_pages);
+      TAILQ_REMOVE(&pmap->pte_pages, pg, pageq);
+      vm_page_free(pg);
+    }
```
[the remark this answers is not in the document]
Comment (bkjg) [reply sys/aarch64/pmap.c]: Okay, we don't need 'cause we are removing this pmap. I fixed it.

File: sys/mips/pmap.c [hunk 3]
```diff
@@ -347,9 +347,10 @@ void pmap_remove(pmap_t *pmap, vaddr_t start, vaddr_t end) {
       paddr_t pa;
       if (pmap_extract_nolock(pmap, va, &pa)) {
         vm_page_t *pg = vm_page_find(pa);
-        WITH_MTX_LOCK (pv_list_lock)
+        WITH_MTX_LOCK (pv_list_lock) {
           pv_remove(pmap, va, pg);
-        pmap_pte_write(pmap, va, empty_pte(pmap), 0);
+          pmap_pte_write(pmap, va, empty_pte(pmap), 0);
```
[the remark this answers is not in the document]
Comment (bkjg) [reply sys/mips/pmap.c]: This code was deleted. pv_list_lock is not needed here.

**Comment by cahirwpz** [comment]
Since the PR depends on #832 that was closed, we are not going to merge this one as well.

**Pull request closed without merging**